Base/usr/share/man/man1/syscall.md

```diff
@@ -0,0 +1,58 @@
+## Name
+
+syscall - test a system call
+
+## Synopsis
+
+`syscall [-o] [-l] [-h] entry [args; buf==BUFSIZ buffer]`
+
+## Description
+
+syscall(1) is an effort to port the plan9 syscall command to Serenity. It can invoke a system call with the given arguments from the command line.
+
+## Options
+
+* `-o`: Output to stdout the contents of the buffer argument specified as buf.
```

Comment:
(English is not my ~favorite~ native language, but my gut feeling insists that) "to stdout" goes at the end:
> Output the contents of the buffer argument specified as buf to stdout.

`buf` should be italic btw, as it's an option.

Reply:
Question: when would it make sense *not* to do that?

Reply:
Maybe if you are executing the system call and only care about the return value and not the actual buffer contents.

Reply:
In that case why don't you just send stdout to `/dev/null`?

Reply:
Yeah, that is possible, but currently the return status is just printed to stdout, we can change the utility to use the syscall return value as the exit status.

Reply:
Doesn't it use `perror()`? `perror()` outputs (or should output) to stderr, not stdout.

Reply:
Yeah sure, you're right. We should use the invoked syscall return value as the program's exit status and then always print the buffer contents? Is that bad practice?

Reply:
> We should use the invoked syscall return value as the program's exit status

I wouldn't do that directly; `-ENOENT` or somesuch makes for a weird exit status 😀

Let's just exit with 0 or 1 depending on whether the syscall succeeded or not, and `perror()` the error in the latter case.

And always copy out the buffer to stdout. Don't use `printf("%s")` for that, by the way — the buffer is an arbitrary byte sequence which doesn't have to be a C null-terminated string (it can actually include 0 bytes in the middle). Use `fwrite()` or something.

Reply:
Thanks for the feedback :)